# 12-17-21 Analysis of regenie burden test for Blood Presure traits

## Create the anno_file, set_list_file and mask_files necessary for burden test

In [1]:
burden_dir=~/UKBiobank/results/REGENIE_results/results_burden_exome/101321_burden_files
anno_sbatch_burden=~/UKBiobank/results/REGENIE_results/results_burden_exome/101321_burden_files/ukb23155_200Kexomes_burdenfiles_$(date +"%Y-%m-%d").sbatch
annotated_file_hg38=~/UKBiobank/results/ukb23155_200Kexomes_annovar/2021_10_12_hg38_exome/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.gz
bim_name=~/UKBiobank/results/ukb23155_200Kexomes_annovar/091321_exome_bim_merge/ukb23155_chr1_chr22_091321.bim
job_size=1
name_prefix='ukb23155_chr1_chr22_qcexomedata_101321_burden_files'
anno_sos=~/project/bioworkflows/variant-annotation/annovar.ipynb
tpl_file=~/project/bioworkflows/admin/csg.yml
container_annovar=$HOME/containers/gatk4-annovar.sif

anno_args="""burden_files
    --cwd $burden_dir
    --annotated_file $annotated_file_hg38
    --bim_name $bim_name
    --name_prefix $name_prefix
    --job_size $job_size
    --container_annovar $container_annovar
    --container_lmm /mnt/mfs/statgen/containers/lmm.sif
"""

sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \
    --template-file $tpl_file \
    --workflow-file $anno_sos \
    --to-script $anno_sbatch_burden\
    --args "$anno_args"


INFO: Running csg: Configuration for Columbia csg partition cluster
INFO: csg is completed.
INFO: csg output:   /home/gl2776/UKBiobank/results/REGENIE_results/results_burden_exome/101321_burden_files/ukb23155_200Kexomes_burdenfiles_2021-10-13.sbatch
INFO: Workflow csg (ID=w2360ed87c5992d95) is executed successfully with 1 completed step.



## Regenie Burden with 200K exomes white British only
### 121521 Systolic Blood Presure

In [1]:
lmm_dir_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/121521_systolicBP_british
lmm_sbatch_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/121521_systolicBP/systolicBP_200k_exomes-regenie-burden_$(date +"%Y-%m-%d").sbatch
phenoFile=~/UKBiobank/phenotype_files/pleiotropy/UKBCauc_cholesterolandbloodpressurefields_inverseranknorm_covariatesage_sex_alcohol_smokingpackyears_pcs
covarFile=~/UKBiobank/phenotype_files/pleiotropy/UKBCauc_cholesterolandbloodpressurefields_inverseranknorm_covariatesage_sex_alcohol_smokingpackyears_pcs
phenoCol=systolic_plus15inmedicated_inverserank
covarCol=SEX
qCovarCol='AGE alcohol1 alcohol2 alcohol3 alcohol4 smoking_dummy1 smoking_dummy2 SBP_PC1 SBP_PC2 SBP_PC3 SBP_PC4 SBP_PC5 SBP_PC6 SBP_PC7 SBP_PC8 SBP_PC9 SBP_PC10' 
genoFile=`echo ~/UKBiobank/data/exome_files/project_VCF/072721_run/plink/ukb23156_c{1..22}.merged.filtered.bed`
bfile=~/UKBiobank/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed
anno_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.anno_file
set_list=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.set_list_file
mask_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.mask_file
aaf_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.aff_file
build_mask=max
aaf_bins='0.005 0.01'
tpl_file=~/project/bioworkflows/admin/csg.yml
lmm_sos=~/project/bioworkflows/GWAS/LMM.ipynb
container_marp=~/containers/marp.sif
container_lmm=~/containers/lmm.sif 
lmm_job_size=1
ylim=20
k=10
reverse_log_p=True
numThreads=20
formatFile_regenie=~/project/UKBB_GWAS_dev/data/regenie_template.yml
bsize=1000
## Trait leave empty for qt traits
trait=
minMAC=1
snpannofile=~/UKBiobank/results/ukb23155_200Kexomes_annovar/2021_10_12_hg38_exome/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.gz

lmm_args="""regenie_burden
    --cwd $lmm_dir_regenie 
    --bfile $bfile 
    --genoFile $genoFile
    --phenoFile $phenoFile 
    --formatFile $formatFile_regenie 
    --phenoCol $phenoCol
    --covarCol $covarCol  
    --qCovarCol $qCovarCol
    --bsize $bsize
    --trait $trait
    --anno_file $anno_file
    --set_list $set_list
    --mask_file $mask_file
    --aaf_file $aaf_file
    --aaf_bins $aaf_bins
    --build_mask $build_mask
    --job_size $lmm_job_size
    --ylim $ylim
    --k $k
    --reverse_log_p $reverse_log_p
    --numThreads $numThreads
    --minMAC $minMAC
    --snpannofile $snpannofile
    --container_lmm $container_lmm
    --container_marp $container_marp
"""

sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \
    --template-file $tpl_file \
    --workflow-file $lmm_sos \
    --to-script $lmm_sbatch_regenie \
    --args "$lmm_args"

/home/dmc2245/miniconda3/lib/python3.8/site-packages/sos_notebook/kernel.py:1334: DeprecationWarning: Kernel._parent_header is deprecated in ipykernel 6. Use .get_parent()
  msg['msg_id'] = self._parent_header['header']['msg_id']


INFO: Running csg: Configuration for Columbia csg partition cluster
INFO: csg is completed.
INFO: csg output:   /home/dmc2245/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/121521_systolicBP/systolicBP_200k_exomes-regenie-burden_2021-12-16.sbatch
INFO: Workflow csg (ID=w6b2e261b0836f1af) is executed successfully with 1 completed step.


### 121521 Diastolic Blood Presure

In [2]:
lmm_dir_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/121521_diastolicBP_british
lmm_sbatch_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/121521_diastolicBP/diastolicBP_200k_exomes-regenie-burden_$(date +"%Y-%m-%d").sbatch
phenoFile=~/UKBiobank/phenotype_files/pleiotropy/UKBCauc_cholesterolandbloodpressurefields_inverseranknorm_covariatesage_sex_alcohol_smokingpackyears_pcs
covarFile=~/UKBiobank/phenotype_files/pleiotropy/UKBCauc_cholesterolandbloodpressurefields_inverseranknorm_covariatesage_sex_alcohol_smokingpackyears_pcs
phenoCol=diastolic_plus10inmedicated_inverserank
covarCol=SEX
qCovarCol='AGE alcohol1 alcohol2 alcohol3 alcohol4 smoking_dummy1 smoking_dummy2 DBP_PC1 DBP_PC2 DBP_PC3 DBP_PC4 DBP_PC5 DBP_PC6 DBP_PC7 DBP_PC8 DBP_PC9 DBP_PC10' 
genoFile=`echo ~/UKBiobank/data/exome_files/project_VCF/072721_run/plink/ukb23156_c{1..22}.merged.filtered.bed`
bfile=~/UKBiobank/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed
anno_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.anno_file
set_list=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.set_list_file
mask_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.mask_file
aaf_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.aff_file
build_mask=max
aaf_bins='0.005 0.01'
tpl_file=~/project/bioworkflows/admin/csg.yml
lmm_sos=~/project/bioworkflows/GWAS/LMM.ipynb
container_marp=~/containers/marp.sif
container_lmm=~/containers/lmm.sif 
lmm_job_size=1
ylim=20
k=10
reverse_log_p=True
numThreads=20
formatFile_regenie=~/project/UKBB_GWAS_dev/data/regenie_template.yml
bsize=1000
trait=
minMAC=1
snpannofile=~/UKBiobank/results/ukb23155_200Kexomes_annovar/2021_10_12_hg38_exome/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.gz

lmm_args="""regenie_burden
    --cwd $lmm_dir_regenie 
    --bfile $bfile 
    --genoFile $genoFile
    --phenoFile $phenoFile 
    --formatFile $formatFile_regenie 
    --phenoCol $phenoCol
    --covarCol $covarCol  
    --qCovarCol $qCovarCol
    --bsize $bsize
    --trait $trait
    --anno_file $anno_file
    --set_list $set_list
    --mask_file $mask_file
    --aaf_file $aaf_file
    --aaf_bins $aaf_bins
    --build_mask $build_mask
    --job_size $lmm_job_size
    --ylim $ylim
    --k $k
    --reverse_log_p $reverse_log_p
    --numThreads $numThreads
    --minMAC $minMAC
    --snpannofile $snpannofile
    --container_lmm $container_lmm
    --container_marp $container_marp
"""

sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \
    --template-file $tpl_file \
    --workflow-file $lmm_sos \
    --to-script $lmm_sbatch_regenie \
    --args "$lmm_args"

INFO: Running csg: Configuration for Columbia csg partition cluster
INFO: csg is completed.
INFO: csg output:   /home/dmc2245/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/121521_diastolicBP/diastolicBP_200k_exomes-regenie-burden_2021-12-16.sbatch
INFO: Workflow csg (ID=wc620d9e819673c30) is executed successfully with 1 completed step.


## Create the genotypeFile 

In [ ]:
cwd=~/UKBiobank/genotype_files_processed/010622_exp_white_finalqc
genoFile=~/UKBiobank_Yale_transfer/pleiotropy_geneticfiles/UKB_originalgenotypefilesdownloaded083019/UKB_genotypedatadownloaded083019.bed 
keep_samples=~/UKBiobank/genotype_files_processed/010622_exp_white_exomed_mind0.1/cache/UKB_genotypedatadownloaded083019.genotype_files_processed.filtered.extracted.010622_exp_white_exomed_mind0.1.filtered.pass_qc.samplelist
keep_variants=~/UKBiobank/genotype_files_processed/010622_exp_white_exomed_mind0.1/cache/UKB_genotypedatadownloaded083019.genotype_files_processed.filtered.extracted.010622_exp_white_exomed_mind0.1.filtered.010622_exp_white_exomed_mind0.1.filtered.bim
maf_filter=0
geno_filter=0
hwe_filter=0
mind_filter=0
mem='30G'
gwasqc_sos=~/project/bioworkflows/GWAS/GWAS_QC.ipynb
gwasqc_sbatch=~/pleiotropy/LMM/gwasqc1_extwhite_$(date +"%Y-%m-%d").sbatch
container_lmm=~/containers/lmm.sif
tpl_file=~/project/bioworkflows/admin/csg.yml
numThreads=1
job_size=1

gwasqc1_args="""qc:1
    --cwd $cwd
    --genoFile $genoFile
    --keep_samples $keep_samples
    --keep_variants $keep_variants
    --maf_filter $maf_filter
    --geno_filter $geno_filter
    --hwe_filter $hwe_filter
    --mind_filter $mind_filter
    --numThreads $numThreads 
    --job_size $job_size
    --container_lmm $container_lmm
    --mem $mem
"""

sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \
    --template-file $tpl_file \
    --workflow-file $gwasqc_sos \
    --to-script $gwasqc_sbatch \
    --args "$gwasqc1_args"

## Calculate the first 10 PC's

## Systolic and Diastolic 200K

### Step 1

In [ ]:
## Columbia's cluster
cwd=$UKBB_PATH/results/083021_PCA_results/091421_f3393_50Kexomes_pca
gwas_sbatch=$OUT_PATH/qc1_f3393_qcarray_50K_$(date +"%Y-%m-%d").sbatch
## Use qc'ed genotype array
genoFile=~/UKBiobank/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed
keep_samples=$UKBB_PATH/results/083021_PCA_results/090321_UKBB_Hearing_aid_f3393_expandedwhite_24189ind_50Kexomes.keep_id
#Keep variants after LD pruning
keep_variants=$UKBB_PATH/results/083021_PCA_results/090221_ldprun_unrelated/cache/*.090221_ldprun_unrelated.filtered.prune.in

#GWAS QC variables set all of this variables to 0 to avoid doing more filtering
maf_filter=0
geno_filter=0
hwe_filter=0
mind_filter=0
numThreads=1
mem='30G'

gwasqc_args="""qc:1
    --cwd $cwd
    --genoFile $genoFile
    --keep_samples $keep_samples
    --keep_variants $keep_variants
    --maf_filter $maf_filter
    --geno_filter $geno_filter
    --hwe_filter $hwe_filter
    --mind_filter $mind_filter
    --numThreads $numThreads 
    --job_size $job_size
    --container_lmm $container_lmm
    --mem $mem
"""

sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \
    --template-file $tpl_file \
    --workflow-file $gwasqc_sos \
    --to-script $gwas_sbatch \
    --args "$gwasqc_args"

In [3]:
## Columbia's cluster
cwd=~/UKBiobank/genotype_files_processed/010622_exp_white_finalqc/012421_PC_1_10
#This is the bfile obtained in step 1
genoFile=~/UKBiobank/genotype_files_processed/010622_exp_white_finalqc/cache/UKB_genotypedatadownloaded083019.010622_exp_white_finalqc.filtered.extracted.bed
# Format fam file
phenoFile=~/UKBiobank/genotype_files_processed/010622_exp_white_finalqc/cache/UKB_genotypedatadownloaded083019.010622_exp_white_finalqc.filtered.extracted.fam
pca_sbatch=~/pleiotropy/LMM/flashpca_systolicBP_pc10_ext_white_$(date +"%Y-%m-%d").sbatch
k=10
# Pipeline
pca_sos=~/project/bioworkflows/GWAS/PCA.ipynb
numThreads=1
job_size=1
container_lmm=$HOME/containers/lmm.sif
tpl_file=~/project/bioworkflows/admin/csg.yml

pca_args="""flashpca
    --cwd $cwd
    --genoFile $genoFile
    --phenoFile $phenoFile
    --k $k
    --numThreads $numThreads 
    --job_size $job_size
    --container_lmm $container_lmm
"""

sos run  ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\
    --template-file $tpl_file \
    --workflow-file $pca_sos \
    --to-script $pca_sbatch \
    --args "$pca_args"

INFO: Running csg: Configuration for Columbia csg partition cluster
INFO: csg is completed.
INFO: csg output:   /home/dmc2245/pleiotropy/LMM/flashpca_systolicBP_pc10_ext_white_2022-01-24.sbatch
INFO: Workflow csg (ID=wec037a691651b09d) is executed successfully with 1 completed step.


In [4]:
pheno <- read.table("~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_white_bloodpressure_invnorm_pcs", header=T)

In [19]:
head(pheno)
nrow(pheno)

FID     IID     age sex smoking1 smoking2 Alcohol1 Alcohol2 Alcohol3 Alcohol4
1 1000019 1000019 47  0   0        0        0        0        1        0       
2 1000035 1000035 63  1   0        0        0        0        0        0       
3 1000078 1000078 52  0   0        0        1        0        0        0       
4 1000081 1000081 67  1   1        0        1        0        0        0       
5 1000198 1000198 41  0   1        0        1        0        0        0       
6 1000210 1000210 66  1   1        0        0        0        0        0       
  ... DBP_PC12 DBP_PC13 DBP_PC14 DBP_PC15 DBP_PC16 DBP_PC17 DBP_PC18 DBP_PC19
1 ... -0.0010  -0.0019  -0.0006   0.0010   0.0004  -0.0012  -0.0020   0.0003 
2 ... -0.0012   0.0006   0.0011  -0.0008  -0.0001   0.0009   0.0012   0.0010 
3 ...  0.0005   0.0013  -0.0004  -0.0010   0.0001  -0.0026   0.0077  -0.0030 
4 ...  0.0009  -0.0075   0.0046  -0.0068   0.0022  -0.0078  -0.0114   0.0054 
5 ... -0.0025   0.0021   0.0010  -0.0025   0.0003  -0.0015   0.0006   0.0004 
6 ... -0.0015  -0.0001   0.0000  -0.0005   0.0000   0.0000   0.0009  -0.0011 
  DBP_PC20 DBP_status
1  0.0036  Case      
2 -0.0005  Case      
3  0.0010  Case      
4 -0.0117  Case      
5  0.0005  Case      
6 -0.0028  Case

[1] 187908

In [6]:
sys <- pheno[is.na(pheno$systolic_plus15inmedicated_inverserank),] 

In [13]:
dias <- pheno[is.na(pheno$diastolic_plus10inmedicated_inverserank),] 

In [15]:
library(dplyr)
sys %>% select(FID, systolic_plus15inmedicated_inverserank)
nrow(sys)

FID     systolic_plus15inmedicated_inverserank
75     1001825 NA                                    
3262   1085950 NA                                    
3893   1102904 NA                                    
3977   1105106 NA                                    
5950   1158287 NA                                    
7351   1195805 NA                                    
8058   1215530 NA                                    
9419   1251517 NA                                    
10103  1269481 NA                                    
11156  1297404 NA                                    
12737  1339173 NA                                    
13226  1352236 NA                                    
14112  1374729 NA                                    
14271  1379340 NA                                    
17153  1455532 NA                                    
19274  1512486 NA                                    
19473  1517737 NA                                    
19532  1519006 NA                                    
20867  1555527 NA                                    
21670  1576837 NA                                    
22633  1602205 NA                                    
23232  1618334 NA                                    
25045  1667370 NA                                    
26003  1692437 NA                                    
29790  1794793 NA                                    
30798  1820654 NA                                    
31148  1830421 NA                                    
32848  1875633 NA                                    
36087  1964650 NA                                    
38438  2027172 NA                                    
...    ...     ...                                   
155011 5145065 NA                                    
156936 5197627 NA                                    
159703 5270913 NA                                    
165259 5418565 NA                                    
165868 5434448 NA                                    
166089 5440641 NA                                    
166214 5444074 NA                                    
169082 5519758 NA                                    
173566 5639855 NA                                    
174092 5654162 NA                                    
174514 5665658 NA                                    
175271 5686740 NA                                    
175720 5698907 NA                                    
176279 5713428 NA                                    
176281 5713452 NA                                    
176296 5713946 NA                                    
177249 5739159 NA                                    
177968 5759006 NA                                    
178859 5782175 NA                                    
179529 5800001 NA                                    
179559 5800860 NA                                    
181145 5842980 NA                                    
181900 5863365 NA                                    
182743 5886281 NA                                    
182765 5886897 NA                                    
183155 5897311 NA                                    
183225 5899100 NA                                    
184898 5944450 NA                                    
185490 5961332 NA                                    
185989 5974911 NA

[1] 171

In [16]:
dias %>% select(FID, diastolic_plus10inmedicated_inverserank)
nrow(dias)

FID     diastolic_plus10inmedicated_inverserank
75     1001825 NA                                     
3262   1085950 NA                                     
3893   1102904 NA                                     
3977   1105106 NA                                     
5950   1158287 NA                                     
7351   1195805 NA                                     
8058   1215530 NA                                     
9419   1251517 NA                                     
10103  1269481 NA                                     
11156  1297404 NA                                     
12737  1339173 NA                                     
13226  1352236 NA                                     
14112  1374729 NA                                     
14271  1379340 NA                                     
17153  1455532 NA                                     
19274  1512486 NA                                     
19473  1517737 NA                                     
19532  1519006 NA                                     
20867  1555527 NA                                     
21670  1576837 NA                                     
22633  1602205 NA                                     
23232  1618334 NA                                     
25045  1667370 NA                                     
26003  1692437 NA                                     
29790  1794793 NA                                     
30798  1820654 NA                                     
31148  1830421 NA                                     
32848  1875633 NA                                     
36087  1964650 NA                                     
38438  2027172 NA                                     
...    ...     ...                                    
155011 5145065 NA                                     
156936 5197627 NA                                     
159703 5270913 NA                                     
165259 5418565 NA                                     
165868 5434448 NA                                     
166089 5440641 NA                                     
166214 5444074 NA                                     
169082 5519758 NA                                     
173566 5639855 NA                                     
174092 5654162 NA                                     
174514 5665658 NA                                     
175271 5686740 NA                                     
175720 5698907 NA                                     
176279 5713428 NA                                     
176281 5713452 NA                                     
176296 5713946 NA                                     
177249 5739159 NA                                     
177968 5759006 NA                                     
178859 5782175 NA                                     
179529 5800001 NA                                     
179559 5800860 NA                                     
181145 5842980 NA                                     
181900 5863365 NA                                     
182743 5886281 NA                                     
182765 5886897 NA                                     
183155 5897311 NA                                     
183225 5899100 NA                                     
184898 5944450 NA                                     
185490 5961332 NA                                     
185989 5974911 NA

[1] 171

In [18]:
all_equal(sys,dias)

[1] TRUE

## 02-16-22 GWAS burden analysis for extended white using the correct bed file and PC calculation

* Samples=187,908
* Variants=522,952

### Systolic burden

In [ ]:
lmm_dir_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/021622_systolicBP_extwhite
lmm_sbatch_regenie=~/pleiotropy/LMM/systolicBP_200k_exomes-regenie-burden_$(date +"%Y-%m-%d").sbatch
phenoFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_white_bloodpressure_invnorm_pcs_new
covarFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_white_bloodpressure_invnorm_pcs_new
phenoCol=systolic_plus15inmedicated_inverserank
covarCol=sex
qCovarCol='age Alcohol1 Alcohol2 Alcohol3 Alcohol4 smoking1 smoking2 SBP_PC1 SBP_PC2 SBP_PC3 SBP_PC4 SBP_PC5 SBP_PC6 SBP_PC7 SBP_PC8 SBP_PC9 SBP_PC10' 
genoFile=`echo ~/UKBiobank/data/exome_files/project_VCF/072721_run/plink/ukb23156_c{1..22}.merged.filtered.bed`
bfile=~/UKBiobank/genotype_files_processed/010622_exp_white_finalqc/cache/*.bed
anno_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.anno_file
set_list=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.set_list_file
mask_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.mask_file
aaf_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.aff_file
build_mask=max
aaf_bins='0.005 0.01'
tpl_file=~/project/bioworkflows/admin/csg.yml
lmm_sos=~/project/bioworkflows/GWAS/LMM.ipynb
container_marp=~/containers/marp.sif
container_lmm=~/containers/lmm.sif 
lmm_job_size=1
ylim=20
k=10
reverse_log_p=True
numThreads=20
formatFile_regenie=~/project/UKBB_GWAS_dev/data/regenie_template.yml
bsize=1000
## Trait leave empty for qt traits
trait=
minMAC=1
snpannofile=~/UKBiobank/results/ukb23155_200Kexomes_annovar/2021_10_12_hg38_exome/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.gz

lmm_args="""regenie_burden
    --cwd $lmm_dir_regenie 
    --bfile $bfile 
    --genoFile $genoFile
    --phenoFile $phenoFile 
    --formatFile $formatFile_regenie 
    --phenoCol $phenoCol
    --covarCol $covarCol  
    --qCovarCol $qCovarCol
    --bsize $bsize
    --trait $trait
    --anno_file $anno_file
    --set_list $set_list
    --mask_file $mask_file
    --aaf_file $aaf_file
    --aaf_bins $aaf_bins
    --build_mask $build_mask
    --job_size $lmm_job_size
    --ylim $ylim
    --k $k
    --reverse_log_p $reverse_log_p
    --numThreads $numThreads
    --minMAC $minMAC
    --snpannofile $snpannofile
    --container_lmm $container_lmm
    --container_marp $container_marp
"""

sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \
    --template-file $tpl_file \
    --workflow-file $lmm_sos \
    --to-script $lmm_sbatch_regenie \
    --args "$lmm_args"

### Diastolic BP burden

In [ ]:
lmm_dir_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/021622_diastolicBP_extwhite
lmm_sbatch_regenie=~/pleiotropy/LMM/diastolicBP_200k_exomes-regenie-burden_$(date +"%Y-%m-%d").sbatch
phenoFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_white_bloodpressure_invnorm_pcs_new
covarFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_white_bloodpressure_invnorm_pcs_new
phenoCol=diastolic_plus10inmedicated_inverserank
covarCol=sex
qCovarCol='age Alcohol1 Alcohol2 Alcohol3 Alcohol4 smoking1 smoking2 DBP_PC1 DBP_PC2 DBP_PC3 DBP_PC4 DBP_PC5 DBP_PC6 DBP_PC7 DBP_PC8 DBP_PC9 DBP_PC10' 
genoFile=`echo ~/UKBiobank/data/exome_files/project_VCF/072721_run/plink/ukb23156_c{1..22}.merged.filtered.bed`
bfile=~/UKBiobank/genotype_files_processed/010622_exp_white_finalqc/cache/*.bed
anno_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.anno_file
set_list=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.set_list_file
mask_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.mask_file
aaf_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.aff_file
build_mask=max
aaf_bins='0.005 0.01'
tpl_file=~/project/bioworkflows/admin/csg.yml
lmm_sos=~/project/bioworkflows/GWAS/LMM.ipynb
container_marp=~/containers/marp.sif
container_lmm=~/containers/lmm.sif 
lmm_job_size=1
ylim=20
k=10
reverse_log_p=True
numThreads=20
formatFile_regenie=~/project/UKBB_GWAS_dev/data/regenie_template.yml
bsize=1000
trait=
minMAC=1
snpannofile=~/UKBiobank/results/ukb23155_200Kexomes_annovar/2021_10_12_hg38_exome/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.gz

lmm_args="""regenie_burden
    --cwd $lmm_dir_regenie 
    --bfile $bfile 
    --genoFile $genoFile
    --phenoFile $phenoFile 
    --formatFile $formatFile_regenie 
    --phenoCol $phenoCol
    --covarCol $covarCol  
    --qCovarCol $qCovarCol
    --bsize $bsize
    --trait $trait
    --anno_file $anno_file
    --set_list $set_list
    --mask_file $mask_file
    --aaf_file $aaf_file
    --aaf_bins $aaf_bins
    --build_mask $build_mask
    --job_size $lmm_job_size
    --ylim $ylim
    --k $k
    --reverse_log_p $reverse_log_p
    --numThreads $numThreads
    --minMAC $minMAC
    --snpannofile $snpannofile
    --container_lmm $container_lmm
    --container_marp $container_marp
"""

sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \
    --template-file $tpl_file \
    --workflow-file $lmm_sos \
    --to-script $lmm_sbatch_regenie \
    --args "$lmm_args"

## 02-17-22 GWAS analysis for BP traits  - Single Variant Analysis with Exome data

* Samples=187,908
* Variants=522,952


### SBP

In [5]:
## All filters set to 0 because the version of the bfile has already been QC'ed previously and there is not need to do it here
maf_filter=0
geno_filter=0
hwe_filter=0
mind_filter=0
lmm_dir_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_exome_data/021722_systolicBP_extwhite/
lmm_sbatch_regenie=~/pleiotropy/LMM/systolicBP_200k_exomes-regenie_extwhite_$(date +"%Y-%m-%d").sbatch
phenoFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_white_bloodpressure_invnorm_pcs_new
covarFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_white_bloodpressure_invnorm_pcs_new
phenoCol=systolic_plus15inmedicated_inverserank
covarCol=sex
qCovarCol='age Alcohol1 Alcohol2 Alcohol3 Alcohol4 smoking1 smoking2 SBP_PC1 SBP_PC2 SBP_PC3 SBP_PC4 SBP_PC5 SBP_PC6 SBP_PC7 SBP_PC8 SBP_PC9 SBP_PC10' 
# Use the QC'ed exome files variant and sample missingness < 10%
genoFile=`echo ~/UKBiobank/data/exome_files/project_VCF/072721_run/plink/ukb23156_c{1..22}.merged.filtered.bed`
#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08
bfile=~/UKBiobank/genotype_files_processed/010622_exp_white_finalqc/cache/*.bed
tpl_file=~/project/bioworkflows/admin/csg.yml
lmm_sos=~/project/bioworkflows/GWAS/LMM.ipynb
container_marp=~/containers/marp.sif
container_lmm=~/containers/lmm.sif 
lmm_job_size=1
ylim=0
reverse_log_p=True
numThreads=20
formatFile_regenie=~/project/UKBB_GWAS_dev/data/regenie_template.yml
bsize=1000
trait=
## Using this MAC the default in regenie analysis
minMAC=5
label_annotate='SNP'

## After doing the annotation you can create the annotatted manhatan plot
#anno_file=$lmm_exome_dir_regenie/090921_f3393_hearing_aid_200K/091321_annotation/*.formatted.csv
#label_annotate=Gene
#known_vars=
#new_vars=
# If --annotate then it will add the label to the plot otherwise --no-annotate

lmm_args="""regenie
    --cwd $lmm_dir_regenie 
    --bfile $bfile 
    --genoFile $genoFile
    --phenoFile $phenoFile 
    --formatFile $formatFile_regenie 
    --phenoCol $phenoCol
    --covarCol $covarCol  
    --qCovarCol $qCovarCol
    --bsize $bsize
    --lowmem_dir $lowmem_dir
    --trait $trait 
    --bgenMinMAF $bgenMinMAF 
    --bgenMinINFO $bgenMinINFO
    --maf_filter $maf_filter
    --geno_filter $geno_filter
    --hwe_filter $hwe_filter
    --mind_filter $mind_filter
    --minMAC $minMAC
    --job_size $lmm_job_size
    --ylim $ylim
    --reverse_log_p $reverse_log_p
    --label_annotate $label_annotate
    --numThreads $numThreads
    --container_lmm $container_lmm
    --container_marp $container_marp
"""

sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\
    --template-file $tpl_file \
    --workflow-file $lmm_sos \
    --to-script $lmm_sbatch_regenie \
    --args "$lmm_args" 

INFO: Running csg: Configuration for Columbia csg partition cluster
INFO: csg is completed.
INFO: csg output:   /home/dmc2245/pleiotropy/LMM/systolicBP_200k_exomes-regenie_extwhite_2022-02-17.sbatch
INFO: Workflow csg (ID=wb628f2e6fd7ff0a0) is executed successfully with 1 completed step.



### DBP

In [6]:
## All filters set to 0 because the version of the bfile has already been QC'ed previously and there is not need to do it here
maf_filter=0
geno_filter=0
hwe_filter=0
mind_filter=0
lmm_dir_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_exome_data/021722_diastolicBP_extwhite/
lmm_sbatch_regenie=~/pleiotropy/LMM/diastolicBP_200k_exomes-regenie_extwhite_$(date +"%Y-%m-%d").sbatch
phenoFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_white_bloodpressure_invnorm_pcs_new
phenoCol=diastolic_plus10inmedicated_inverserank
covarCol=sex
qCovarCol='age Alcohol1 Alcohol2 Alcohol3 Alcohol4 smoking1 smoking2 DBP_PC1 DBP_PC2 DBP_PC3 DBP_PC4 DBP_PC5 DBP_PC6 DBP_PC7 DBP_PC8 DBP_PC9 DBP_PC10' 
# Use the QC'ed exome files variant and sample missingness < 10%
genoFile=`echo ~/UKBiobank/data/exome_files/project_VCF/072721_run/plink/ukb23156_c{1..22}.merged.filtered.bed`
#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08
bfile=~/UKBiobank/genotype_files_processed/010622_exp_white_finalqc/cache/*.bed
tpl_file=~/project/bioworkflows/admin/csg.yml
lmm_sos=~/project/bioworkflows/GWAS/LMM.ipynb
container_marp=~/containers/marp.sif
container_lmm=~/containers/lmm.sif 
lmm_job_size=1
ylim=0
reverse_log_p=True
numThreads=20
formatFile_regenie=~/project/UKBB_GWAS_dev/data/regenie_template.yml
bsize=1000
trait=
minMAC=5
label_annotate='SNP'

## After doing the annotation you can create the annotatted manhatan plot
#anno_file=$lmm_exome_dir_regenie/090921_f3393_hearing_aid_200K/091321_annotation/*.formatted.csv
#label_annotate=Gene
#known_vars=
#new_vars=
# If --annotate then it will add the label to the plot otherwise --no-annotate

lmm_args="""regenie
    --cwd $lmm_dir_regenie 
    --bfile $bfile 
    --genoFile $genoFile
    --phenoFile $phenoFile 
    --formatFile $formatFile_regenie 
    --phenoCol $phenoCol
    --covarCol $covarCol  
    --qCovarCol $qCovarCol
    --bsize $bsize
    --lowmem_dir $lowmem_dir
    --trait $trait 
    --bgenMinMAF $bgenMinMAF 
    --bgenMinINFO $bgenMinINFO
    --maf_filter $maf_filter
    --geno_filter $geno_filter
    --hwe_filter $hwe_filter
    --mind_filter $mind_filter
    --minMAC $minMAC
    --job_size $lmm_job_size
    --ylim $ylim
    --label_annotate $label_annotate
    --reverse_log_p $reverse_log_p
    --numThreads $numThreads
    --container_lmm $container_lmm
    --container_marp $container_marp
"""

sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\
    --template-file $tpl_file \
    --workflow-file $lmm_sos \
    --to-script $lmm_sbatch_regenie \
    --args "$lmm_args" 

INFO: Running csg: Configuration for Columbia csg partition cluster
INFO: csg is completed.
INFO: csg output:   /home/dmc2245/pleiotropy/LMM/diastolicBP_200k_exomes-regenie_extwhite_2022-02-17.sbatch
INFO: Workflow csg (ID=w72fc077e0d9b345f) is executed successfully with 1 completed step.



### Asian SBP RINT

In [1]:
## All filters set to 0 because the version of the bfile has already been QC'ed previously and there is not need to do it here
maf_filter=0
geno_filter=0
hwe_filter=0
mind_filter=0
lmm_dir_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_exome_data/051222_systolicBP_asian/
lmm_sbatch_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_exome_data/051222_systolicBP_asian/systolicBP_200k_exomes-regenie_asian_$(date +"%Y-%m-%d").sbatch
phenoFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_Asian_bloodpressure_invnorm_pcs
phenoCol=systolic_plus15inmedicated_inverserank
covarCol=sex
qCovarCol=`echo age Alcohol{1..4} smoking{1..2} SBP_PC{1..10}`
# Use the QC'ed exome files variant and sample missingness < 10%
genoFile=`echo ~/UKBiobank/data/exome_files/project_VCF/072721_run/plink/ukb23156_c{1..22}.merged.filtered.bed`
#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08
bfile=~/UKBiobank/genotype_files_processed/021722_asian_finalqc_10157samples_444076variants/cache/*.bed
tpl_file=~/project/bioworkflows/admin/csg.yml
lmm_sos=~/project/bioworkflows/GWAS/LMM.ipynb
container_marp=~/containers/marp.sif
container_lmm=~/containers/lmm.sif 
lmm_job_size=1
ylim=0
reverse_log_p=True
numThreads=20
formatFile_regenie=~/project/UKBB_GWAS_dev/data/regenie_template.yml
bsize=1000
trait=
## Using this MAC the default in regenie analysis
minMAC=5
label_annotate='SNP'

## After doing the annotation you can create the annotatted manhatan plot
#anno_file=$lmm_exome_dir_regenie/090921_f3393_hearing_aid_200K/091321_annotation/*.formatted.csv
#label_annotate=Gene
#known_vars=
#new_vars=
# If --annotate then it will add the label to the plot otherwise --no-annotate

lmm_args="""regenie
    --cwd $lmm_dir_regenie 
    --bfile $bfile 
    --genoFile $genoFile
    --phenoFile $phenoFile 
    --formatFile $formatFile_regenie 
    --phenoCol $phenoCol
    --covarCol $covarCol  
    --qCovarCol $qCovarCol
    --bsize $bsize
    --lowmem_dir $lowmem_dir
    --trait $trait 
    --bgenMinMAF $bgenMinMAF 
    --bgenMinINFO $bgenMinINFO
    --maf_filter $maf_filter
    --geno_filter $geno_filter
    --hwe_filter $hwe_filter
    --mind_filter $mind_filter
    --minMAC $minMAC
    --job_size $lmm_job_size
    --ylim $ylim
    --reverse_log_p $reverse_log_p
    --label_annotate $label_annotate
    --numThreads $numThreads
    --container_lmm $container_lmm
    --container_marp $container_marp
"""

sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\
    --template-file $tpl_file \
    --workflow-file $lmm_sos \
    --to-script $lmm_sbatch_regenie \
    --args "$lmm_args" 

/home/dmc2245/miniconda3/lib/python3.8/site-packages/sos_notebook/kernel.py:1334: DeprecationWarning: Kernel._parent_header is deprecated in ipykernel 6. Use .get_parent()
  msg['msg_id'] = self._parent_header['header']['msg_id']


INFO: Running csg: Configuration for Columbia csg partition cluster
INFO: csg is completed.
INFO: csg output:   /home/dmc2245/UKBiobank/results_pleiotropy/REGENIE_results/results_exome_data/051222_systolicBP_asian/systolicBP_200k_exomes-regenie_asian_2022-05-12.sbatch
INFO: Workflow csg (ID=wf8f1405badbdfdcd) is executed successfully with 1 completed step.



### Asian DBP RINT

In [2]:
## All filters set to 0 because the version of the bfile has already been QC'ed previously and there is not need to do it here
maf_filter=0
geno_filter=0
hwe_filter=0
mind_filter=0
lmm_dir_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_exome_data/051222_diastolicBP_asian/
lmm_sbatch_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_exome_data/051222_diastolicBP_asian/diastolicBP_200k_exomes-regenie_asian_$(date +"%Y-%m-%d").sbatch
phenoFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_Asian_bloodpressure_invnorm_pcs
phenoCol=diastolic_plus10inmedicated_inverserank
covarCol=sex
qCovarCol=`echo age Alcohol{1..4} smoking{1..2} DBP_PC{1..10}`
# Use the QC'ed exome files variant and sample missingness < 10%
genoFile=`echo ~/UKBiobank/data/exome_files/project_VCF/072721_run/plink/ukb23156_c{1..22}.merged.filtered.bed`
#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08
bfile=~/UKBiobank/genotype_files_processed/021722_asian_finalqc_10157samples_444076variants/cache/*.bed
tpl_file=~/project/bioworkflows/admin/csg.yml
lmm_sos=~/project/bioworkflows/GWAS/LMM.ipynb
container_marp=~/containers/marp.sif
container_lmm=~/containers/lmm.sif 
lmm_job_size=1
ylim=0
reverse_log_p=True
numThreads=20
formatFile_regenie=~/project/UKBB_GWAS_dev/data/regenie_template.yml
bsize=1000
trait=
## Using this MAC the default in regenie analysis
minMAC=5
label_annotate='SNP'

## After doing the annotation you can create the annotatted manhatan plot
#anno_file=$lmm_exome_dir_regenie/090921_f3393_hearing_aid_200K/091321_annotation/*.formatted.csv
#label_annotate=Gene
#known_vars=
#new_vars=
# If --annotate then it will add the label to the plot otherwise --no-annotate

lmm_args="""regenie
    --cwd $lmm_dir_regenie 
    --bfile $bfile 
    --genoFile $genoFile
    --phenoFile $phenoFile 
    --formatFile $formatFile_regenie 
    --phenoCol $phenoCol
    --covarCol $covarCol  
    --qCovarCol $qCovarCol
    --bsize $bsize
    --lowmem_dir $lowmem_dir
    --trait $trait 
    --bgenMinMAF $bgenMinMAF 
    --bgenMinINFO $bgenMinINFO
    --maf_filter $maf_filter
    --geno_filter $geno_filter
    --hwe_filter $hwe_filter
    --mind_filter $mind_filter
    --minMAC $minMAC
    --job_size $lmm_job_size
    --ylim $ylim
    --reverse_log_p $reverse_log_p
    --label_annotate $label_annotate
    --numThreads $numThreads
    --container_lmm $container_lmm
    --container_marp $container_marp
"""

sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\
    --template-file $tpl_file \
    --workflow-file $lmm_sos \
    --to-script $lmm_sbatch_regenie \
    --args "$lmm_args" 

INFO: Running csg: Configuration for Columbia csg partition cluster
INFO: csg is completed.
INFO: csg output:   /home/dmc2245/UKBiobank/results_pleiotropy/REGENIE_results/results_exome_data/051222_diastolicBP_asian/diastolicBP_200k_exomes-regenie_asian_2022-05-12.sbatch
INFO: Workflow csg (ID=wc0c63a14ea09a39e) is executed successfully with 1 completed step.



### African SBP RINT

In [3]:
## All filters set to 0 because the version of the bfile has already been QC'ed previously and there is not need to do it here
maf_filter=0
geno_filter=0
hwe_filter=0
mind_filter=0
lmm_dir_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_exome_data/051222_systolicBP_african/
lmm_sbatch_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_exome_data/051222_systolicBP_african/systolicBP_200k_exomes-regenie_african_$(date +"%Y-%m-%d").sbatch
phenoFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_African_bloodpressure_invnorm_pcs
phenoCol=systolic_plus15inmedicated_inverserank
covarCol=sex
qCovarCol=`echo age Alcohol{1..4} smoking{1..2} SBP_PC{1..10}`
# Use the QC'ed exome files variant and sample missingness < 10%
genoFile=`echo ~/UKBiobank/data/exome_files/project_VCF/072721_run/plink/ukb23156_c{1..22}.merged.filtered.bed`
#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08
bfile=~/UKBiobank/genotype_files_processed/021722_african_finalqc_8591samples_351690variants/cache/*.bed
tpl_file=~/project/bioworkflows/admin/csg.yml
lmm_sos=~/project/bioworkflows/GWAS/LMM.ipynb
container_marp=~/containers/marp.sif
container_lmm=~/containers/lmm.sif 
lmm_job_size=1
ylim=0
reverse_log_p=True
numThreads=20
formatFile_regenie=~/project/UKBB_GWAS_dev/data/regenie_template.yml
bsize=1000
trait=
## Using this MAC the default in regenie analysis
minMAC=5
label_annotate='SNP'

## After doing the annotation you can create the annotatted manhatan plot
#anno_file=$lmm_exome_dir_regenie/090921_f3393_hearing_aid_200K/091321_annotation/*.formatted.csv
#label_annotate=Gene
#known_vars=
#new_vars=
# If --annotate then it will add the label to the plot otherwise --no-annotate

lmm_args="""regenie
    --cwd $lmm_dir_regenie 
    --bfile $bfile 
    --genoFile $genoFile
    --phenoFile $phenoFile 
    --formatFile $formatFile_regenie 
    --phenoCol $phenoCol
    --covarCol $covarCol  
    --qCovarCol $qCovarCol
    --bsize $bsize
    --lowmem_dir $lowmem_dir
    --trait $trait 
    --bgenMinMAF $bgenMinMAF 
    --bgenMinINFO $bgenMinINFO
    --maf_filter $maf_filter
    --geno_filter $geno_filter
    --hwe_filter $hwe_filter
    --mind_filter $mind_filter
    --minMAC $minMAC
    --job_size $lmm_job_size
    --ylim $ylim
    --reverse_log_p $reverse_log_p
    --label_annotate $label_annotate
    --numThreads $numThreads
    --container_lmm $container_lmm
    --container_marp $container_marp
"""

sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\
    --template-file $tpl_file \
    --workflow-file $lmm_sos \
    --to-script $lmm_sbatch_regenie \
    --args "$lmm_args" 

INFO: Running csg: Configuration for Columbia csg partition cluster
INFO: csg is completed.
INFO: csg output:   /home/dmc2245/UKBiobank/results_pleiotropy/REGENIE_results/results_exome_data/051222_systolicBP_african/systolicBP_200k_exomes-regenie_african_2022-05-12.sbatch
INFO: Workflow csg (ID=w138684d664c0da91) is executed successfully with 1 completed step.



### African DBP RINT

In [4]:
## All filters set to 0 because the version of the bfile has already been QC'ed previously and there is not need to do it here
maf_filter=0
geno_filter=0
hwe_filter=0
mind_filter=0
lmm_dir_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_exome_data/051222_diastolicBP_african/
lmm_sbatch_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_exome_data/051222_diastolicBP_african/diastolicBP_200k_exomes-regenie_african_$(date +"%Y-%m-%d").sbatch
phenoFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_African_bloodpressure_invnorm_pcs
phenoCol=diastolic_plus10inmedicated_inverserank
covarCol=sex
qCovarCol=`echo age Alcohol{1..4} smoking{1..2} DBP_PC{1..10}`
# Use the QC'ed exome files variant and sample missingness < 10%
genoFile=`echo ~/UKBiobank/data/exome_files/project_VCF/072721_run/plink/ukb23156_c{1..22}.merged.filtered.bed`
#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08
bfile=~/UKBiobank/genotype_files_processed/021722_african_finalqc_8591samples_351690variants/cache/*.bed
tpl_file=~/project/bioworkflows/admin/csg.yml
lmm_sos=~/project/bioworkflows/GWAS/LMM.ipynb
container_marp=~/containers/marp.sif
container_lmm=~/containers/lmm.sif 
lmm_job_size=1
ylim=0
reverse_log_p=True
numThreads=20
formatFile_regenie=~/project/UKBB_GWAS_dev/data/regenie_template.yml
bsize=1000
trait=
## Using this MAC the default in regenie analysis
minMAC=5
label_annotate='SNP'

## After doing the annotation you can create the annotatted manhatan plot
#anno_file=$lmm_exome_dir_regenie/090921_f3393_hearing_aid_200K/091321_annotation/*.formatted.csv
#label_annotate=Gene
#known_vars=
#new_vars=
# If --annotate then it will add the label to the plot otherwise --no-annotate

lmm_args="""regenie
    --cwd $lmm_dir_regenie 
    --bfile $bfile 
    --genoFile $genoFile
    --phenoFile $phenoFile 
    --formatFile $formatFile_regenie 
    --phenoCol $phenoCol
    --covarCol $covarCol  
    --qCovarCol $qCovarCol
    --bsize $bsize
    --lowmem_dir $lowmem_dir
    --trait $trait 
    --bgenMinMAF $bgenMinMAF 
    --bgenMinINFO $bgenMinINFO
    --maf_filter $maf_filter
    --geno_filter $geno_filter
    --hwe_filter $hwe_filter
    --mind_filter $mind_filter
    --minMAC $minMAC
    --job_size $lmm_job_size
    --ylim $ylim
    --reverse_log_p $reverse_log_p
    --label_annotate $label_annotate
    --numThreads $numThreads
    --container_lmm $container_lmm
    --container_marp $container_marp
"""

sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\
    --template-file $tpl_file \
    --workflow-file $lmm_sos \
    --to-script $lmm_sbatch_regenie \
    --args "$lmm_args" 

INFO: Running csg: Configuration for Columbia csg partition cluster
INFO: csg is completed.
INFO: csg output:   /home/dmc2245/UKBiobank/results_pleiotropy/REGENIE_results/results_exome_data/051222_diastolicBP_african/diastolicBP_200k_exomes-regenie_african_2022-05-12.sbatch
INFO: Workflow csg (ID=w009464eb0ae3a6eb) is executed successfully with 1 completed step.



## 02-17-22 ASIAN burden analysis

The final bed file consist of this number of samples and variants

* Variants:444076
* Samples: 10157

### Generation of the final bed file that is QC'ed and outiers have been removed

In [2]:
#Columbia's cluster
cwd=~/UKBiobank/genotype_files_processed/021722_asian_finalqc_10157samples_444076variants
# Original bfile containing all of the samples Columbias's cluster
genoFile=~/UKBiobank_Yale_transfer/pleiotropy_geneticfiles/UKB_originalgenotypefilesdownloaded083019/UKB_genotypedatadownloaded083019.bed
#To keep the samples of white individuals only
keep_samples=~/UKBiobank/genotype_files_processed/010622_asian_10189ind/UKB_genotypedatadownloaded083019.010722_sample_var_final_qc.filtered.extracted_ASIAN.id
keep_variants=~/UKBiobank/genotype_files_processed/010622_asian_10189ind/UKB_genotypedatadownloaded083019.010722_sample_var_final_qc.filtered.extracted_ASIAN.snplist
tpl_file=~/project/bioworkflows/admin/csg.yml
maf_filter=0
geno_filter=0
hwe_filter=0
mind_filter=0
mem='30G'

gwasqc_sos=~/UKBB_GWAS_dev/bioworkflows/GWAS/GWAS_QC.ipynb
gwasqc_sbatch=~/pleiotropy/LMM/asian_final_qc_$(date +"%Y-%m-%d").sbatch

gwasqc1_args="""qc:1
    --cwd $cwd
    --genoFile $genoFile
    --keep_samples $keep_samples
    --keep_variants $keep_variants
    --maf_filter $maf_filter
    --geno_filter $geno_filter
    --hwe_filter $hwe_filter
    --mind_filter $mind_filter
    --numThreads $numThreads 
    --job_size $job_size
    --container_lmm $container_lmm
    --mem $mem
"""

sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \
    --template-file $tpl_file \
    --workflow-file $gwasqc_sos \
    --to-script $gwasqc_sbatch \
    --args "$gwasqc1_args"

INFO: Running csg: Configuration for Columbia csg partition cluster
INFO: csg is completed.
INFO: csg output:   /home/dmc2245/pleiotropy/LMM/asian_final_qc_2022-02-17.sbatch
INFO: Workflow csg (ID=w6e4a7b68b24be629) is executed successfully with 1 completed step.



### Systolic BP

In [3]:
lmm_dir_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/021622_systolicBP_asian
lmm_sbatch_regenie=~/pleiotropy/LMM/systolicBP_200k_exomes-regenie-burden_asian$(date +"%Y-%m-%d").sbatch
phenoFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_Asian_bloodpressure_invnorm_pcs
covarFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_Asian_bloodpressure_invnorm_pcs
phenoCol=systolic_plus15inmedicated_inverserank
covarCol=sex
qCovarCol='age Alcohol1 Alcohol2 Alcohol3 Alcohol4 smoking1 smoking2 SBP_PC1 SBP_PC2 SBP_PC3 SBP_PC4 SBP_PC5 SBP_PC6 SBP_PC7 SBP_PC8 SBP_PC9 SBP_PC10' 
genoFile=`echo ~/UKBiobank/data/exome_files/project_VCF/072721_run/plink/ukb23156_c{1..22}.merged.filtered.bed`
bfile=~/UKBiobank/genotype_files_processed/021722_asian_finalqc_10157samples_444076variants/cache/*.bed
anno_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.anno_file
set_list=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.set_list_file
mask_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.mask_file
aaf_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.aff_file
build_mask=max
aaf_bins='0.005 0.01'
tpl_file=~/project/bioworkflows/admin/csg.yml
lmm_sos=~/project/bioworkflows/GWAS/LMM.ipynb
container_marp=~/containers/marp.sif
container_lmm=~/containers/lmm.sif 
lmm_job_size=1
ylim=0
k=10
reverse_log_p=True
numThreads=20
formatFile_regenie=~/project/UKBB_GWAS_dev/data/regenie_template.yml
bsize=1000
## Trait leave empty for qt traits
trait=
minMAC=1
snpannofile=~/UKBiobank/results/ukb23155_200Kexomes_annovar/2021_10_12_hg38_exome/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.gz

lmm_args="""regenie_burden
    --cwd $lmm_dir_regenie 
    --bfile $bfile 
    --genoFile $genoFile
    --phenoFile $phenoFile 
    --formatFile $formatFile_regenie 
    --phenoCol $phenoCol
    --covarCol $covarCol  
    --qCovarCol $qCovarCol
    --bsize $bsize
    --trait $trait
    --anno_file $anno_file
    --set_list $set_list
    --mask_file $mask_file
    --aaf_file $aaf_file
    --aaf_bins $aaf_bins
    --build_mask $build_mask
    --job_size $lmm_job_size
    --ylim $ylim
    --k $k
    --reverse_log_p $reverse_log_p
    --numThreads $numThreads
    --minMAC $minMAC
    --snpannofile $snpannofile
    --container_lmm $container_lmm
    --container_marp $container_marp
"""

sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \
    --template-file $tpl_file \
    --workflow-file $lmm_sos \
    --to-script $lmm_sbatch_regenie \
    --args "$lmm_args"

INFO: Running csg: Configuration for Columbia csg partition cluster
INFO: csg is completed.
INFO: csg output:   /home/dmc2245/pleiotropy/LMM/systolicBP_200k_exomes-regenie-burden_asian2022-02-17.sbatch
INFO: Workflow csg (ID=w9274a47cc1a187b1) is executed successfully with 1 completed step.



### Diastolic BP

In [4]:
lmm_dir_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/021622_diastolicBP_asian
lmm_sbatch_regenie=~/pleiotropy/LMM/diastolicBP_200k_exomes-regenie-burden_asian_$(date +"%Y-%m-%d").sbatch
phenoFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_Asian_bloodpressure_invnorm_pcs
covarFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_Asian_bloodpressure_invnorm_pcs
phenoCol=diastolic_plus10inmedicated_inverserank
covarCol=sex
qCovarCol='age Alcohol1 Alcohol2 Alcohol3 Alcohol4 smoking1 smoking2 DBP_PC1 DBP_PC2 DBP_PC3 DBP_PC4 DBP_PC5 DBP_PC6 DBP_PC7 DBP_PC8 DBP_PC9 DBP_PC10' 
genoFile=`echo ~/UKBiobank/data/exome_files/project_VCF/072721_run/plink/ukb23156_c{1..22}.merged.filtered.bed`
bfile=~/UKBiobank/genotype_files_processed/021722_asian_finalqc_10157samples_444076variants/cache/*.bed
anno_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.anno_file
set_list=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.set_list_file
mask_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.mask_file
aaf_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.aff_file
build_mask=max
aaf_bins='0.005 0.01'
tpl_file=~/project/bioworkflows/admin/csg.yml
lmm_sos=~/project/bioworkflows/GWAS/LMM.ipynb
container_marp=~/containers/marp.sif
container_lmm=~/containers/lmm.sif 
lmm_job_size=1
ylim=0
k=10
reverse_log_p=True
numThreads=20
formatFile_regenie=~/project/UKBB_GWAS_dev/data/regenie_template.yml
bsize=1000
trait=
minMAC=1
snpannofile=~/UKBiobank/results/ukb23155_200Kexomes_annovar/2021_10_12_hg38_exome/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.gz

lmm_args="""regenie_burden
    --cwd $lmm_dir_regenie 
    --bfile $bfile 
    --genoFile $genoFile
    --phenoFile $phenoFile 
    --formatFile $formatFile_regenie 
    --phenoCol $phenoCol
    --covarCol $covarCol  
    --qCovarCol $qCovarCol
    --bsize $bsize
    --trait $trait
    --anno_file $anno_file
    --set_list $set_list
    --mask_file $mask_file
    --aaf_file $aaf_file
    --aaf_bins $aaf_bins
    --build_mask $build_mask
    --job_size $lmm_job_size
    --ylim $ylim
    --k $k
    --reverse_log_p $reverse_log_p
    --numThreads $numThreads
    --minMAC $minMAC
    --snpannofile $snpannofile
    --container_lmm $container_lmm
    --container_marp $container_marp
"""

sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \
    --template-file $tpl_file \
    --workflow-file $lmm_sos \
    --to-script $lmm_sbatch_regenie \
    --args "$lmm_args"

INFO: Running csg: Configuration for Columbia csg partition cluster
INFO: csg is completed.
INFO: csg output:   /home/dmc2245/pleiotropy/LMM/diastolicBP_200k_exomes-regenie-burden_asian_2022-02-17.sbatch
INFO: Workflow csg (ID=wa14c29a01f502d99) is executed successfully with 1 completed step.



# AFRICAN burden analysis

### Generation of the final bed file that is QC'ed and outliers have been removed

In [2]:
#Columbia's cluster
cwd=~/UKBiobank/genotype_files_processed/021722_african_finalqc_8591samples_351690variants
# Original bfile containing all of the samples Columbias's cluster
genoFile=~/UKBiobank_Yale_transfer/pleiotropy_geneticfiles/UKB_originalgenotypefilesdownloaded083019/UKB_genotypedatadownloaded083019.bed
#To keep the samples of white individuals only
keep_samples=~/UKBiobank/genotype_files_processed/010622_african_9096ind/UKB_genotypedatadownloaded083019.010722_sample_var_final_qc.filtered.extracted.id
keep_variants=~/UKBiobank/genotype_files_processed/010622_african_9096ind/UKB_genotypedatadownloaded083019.010722_sample_var_final_qc.filtered.extracted.snplist
tpl_file=~/project/bioworkflows/admin/csg.yml
maf_filter=0
geno_filter=0
hwe_filter=0
mind_filter=0
mem='20G'
job_size=1
numThreads=8
gwasqc_sos=~/project/bioworkflows/GWAS/GWAS_QC.ipynb
gwasqc_sbatch=~/pleiotropy/LMM/african_final_qc_$(date +"%Y-%m-%d").sbatch
container_lmm=~/containers/lmm.sif 

gwasqc1_args="""qc:1
    --cwd $cwd
    --genoFile $genoFile
    --keep_samples $keep_samples
    --keep_variants $keep_variants
    --maf_filter $maf_filter
    --geno_filter $geno_filter
    --hwe_filter $hwe_filter
    --mind_filter $mind_filter
    --numThreads $numThreads 
    --job_size $job_size
    --container_lmm $container_lmm
    --mem $mem
"""

sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \
    --template-file $tpl_file \
    --workflow-file $gwasqc_sos \
    --to-script $gwasqc_sbatch \
    --args "$gwasqc1_args"

INFO: Running csg: Configuration for Columbia csg partition cluster
INFO: csg is completed.
INFO: csg output:   /home/dmc2245/pleiotropy/LMM/african_final_qc_2022-02-18.sbatch
INFO: Workflow csg (ID=wdbb9ef0ed5ed88c9) is executed successfully with 1 completed step.



### Systolic BP

In [3]:
lmm_dir_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/022222_systolicBP_african
lmm_sbatch_regenie=~/pleiotropy/LMM/systolicBP_200k_exomes-regenie-burden_african_$(date +"%Y-%m-%d").sbatch
phenoFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_African_bloodpressure_invnorm_pcs
covarFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_African_bloodpressure_invnorm_pcs
phenoCol=systolic_plus15inmedicated_inverserank
covarCol=sex
qCovarCol='age Alcohol1 Alcohol2 Alcohol3 Alcohol4 smoking1 smoking2 SBP_PC1 SBP_PC2 SBP_PC3 SBP_PC4 SBP_PC5 SBP_PC6 SBP_PC7 SBP_PC8 SBP_PC9 SBP_PC10' 
genoFile=`echo ~/UKBiobank/data/exome_files/project_VCF/072721_run/plink/ukb23156_c{1..22}.merged.filtered.bed`
bfile=~/UKBiobank/genotype_files_processed/021722_african_finalqc_8591samples_351690variants/cache/*.bed
anno_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.anno_file
set_list=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.set_list_file
mask_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.mask_file
aaf_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.aff_file
build_mask=max
aaf_bins='0.005 0.01'
tpl_file=~/project/bioworkflows/admin/csg.yml
lmm_sos=~/project/bioworkflows/GWAS/LMM.ipynb
container_marp=~/containers/marp.sif
container_lmm=~/containers/lmm.sif 
lmm_job_size=1
ylim=0
k=10
reverse_log_p=True
numThreads=20
formatFile_regenie=~/project/UKBB_GWAS_dev/data/regenie_template.yml
bsize=1000
## Trait leave empty for qt traits
trait=
minMAC=1
snpannofile=~/UKBiobank/results/ukb23155_200Kexomes_annovar/2021_10_12_hg38_exome/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.gz

lmm_args="""regenie_burden
    --cwd $lmm_dir_regenie 
    --bfile $bfile 
    --genoFile $genoFile
    --phenoFile $phenoFile 
    --formatFile $formatFile_regenie 
    --phenoCol $phenoCol
    --covarCol $covarCol  
    --qCovarCol $qCovarCol
    --bsize $bsize
    --trait $trait
    --anno_file $anno_file
    --set_list $set_list
    --mask_file $mask_file
    --aaf_file $aaf_file
    --aaf_bins $aaf_bins
    --build_mask $build_mask
    --job_size $lmm_job_size
    --ylim $ylim
    --k $k
    --reverse_log_p $reverse_log_p
    --numThreads $numThreads
    --minMAC $minMAC
    --snpannofile $snpannofile
    --container_lmm $container_lmm
    --container_marp $container_marp
"""

sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \
    --template-file $tpl_file \
    --workflow-file $lmm_sos \
    --to-script $lmm_sbatch_regenie \
    --args "$lmm_args"

INFO: Running csg: Configuration for Columbia csg partition cluster
INFO: csg is completed.
INFO: csg output:   /home/dmc2245/pleiotropy/LMM/systolicBP_200k_exomes-regenie-burden_african_2022-02-22.sbatch
INFO: Workflow csg (ID=wd745d0e7a1f4c5c9) is executed successfully with 1 completed step.



### Diastolic BP

In [4]:
lmm_dir_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/022222_diastolicBP_african
lmm_sbatch_regenie=~/pleiotropy/LMM/diastolicBP_200k_exomes-regenie-burden_african_$(date +"%Y-%m-%d").sbatch
phenoFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_African_bloodpressure_invnorm_pcs
covarFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_African_bloodpressure_invnorm_pcs
phenoCol=diastolic_plus10inmedicated_inverserank
covarCol=sex
qCovarCol='age Alcohol1 Alcohol2 Alcohol3 Alcohol4 smoking1 smoking2 DBP_PC1 DBP_PC2 DBP_PC3 DBP_PC4 DBP_PC5 DBP_PC6 DBP_PC7 DBP_PC8 DBP_PC9 DBP_PC10' 
genoFile=`echo ~/UKBiobank/data/exome_files/project_VCF/072721_run/plink/ukb23156_c{1..22}.merged.filtered.bed`
bfile=~/UKBiobank/genotype_files_processed/021722_african_finalqc_8591samples_351690variants/cache/*.bed
anno_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.anno_file
set_list=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.set_list_file
mask_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.mask_file
aaf_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.aff_file
build_mask=max
aaf_bins='0.005 0.01'
tpl_file=~/project/bioworkflows/admin/csg.yml
lmm_sos=~/project/bioworkflows/GWAS/LMM.ipynb
container_marp=~/containers/marp.sif
container_lmm=~/containers/lmm.sif 
lmm_job_size=1
ylim=0
k=10
reverse_log_p=True
numThreads=20
formatFile_regenie=~/project/UKBB_GWAS_dev/data/regenie_template.yml
bsize=1000
trait=
minMAC=1
snpannofile=~/UKBiobank/results/ukb23155_200Kexomes_annovar/2021_10_12_hg38_exome/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.gz

lmm_args="""regenie_burden
    --cwd $lmm_dir_regenie 
    --bfile $bfile 
    --genoFile $genoFile
    --phenoFile $phenoFile 
    --formatFile $formatFile_regenie 
    --phenoCol $phenoCol
    --covarCol $covarCol  
    --qCovarCol $qCovarCol
    --bsize $bsize
    --trait $trait
    --anno_file $anno_file
    --set_list $set_list
    --mask_file $mask_file
    --aaf_file $aaf_file
    --aaf_bins $aaf_bins
    --build_mask $build_mask
    --job_size $lmm_job_size
    --ylim $ylim
    --k $k
    --reverse_log_p $reverse_log_p
    --numThreads $numThreads
    --minMAC $minMAC
    --snpannofile $snpannofile
    --container_lmm $container_lmm
    --container_marp $container_marp
"""

sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \
    --template-file $tpl_file \
    --workflow-file $lmm_sos \
    --to-script $lmm_sbatch_regenie \
    --args "$lmm_args"

INFO: Running csg: Configuration for Columbia csg partition cluster
INFO: csg is completed.
INFO: csg output:   /home/dmc2245/pleiotropy/LMM/diastolicBP_200k_exomes-regenie-burden_african_2022-02-22.sbatch
INFO: Workflow csg (ID=w592764c03e7ebef7) is executed successfully with 1 completed step.



# 04-29-22 Burden analysis with the raw systolic_plus15inmedicated for extended white

In [1]:
lmm_dir_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/042922_systolic_plus15inmedicated_extwhite
lmm_sbatch_regenie=~/pleiotropy/LMM/systolic_plus15inmedicated_200k_exomes-regenie-burden_$(date +"%Y-%m-%d").sbatch
phenoFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_white_bloodpressure_invnorm_pcs_new
covarFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_white_bloodpressure_invnorm_pcs_new
phenoCol=systolic_plus15inmedicated
covarCol=sex
qCovarCol='age Alcohol1 Alcohol2 Alcohol3 Alcohol4 smoking1 smoking2 SBP_PC1 SBP_PC2 SBP_PC3 SBP_PC4 SBP_PC5 SBP_PC6 SBP_PC7 SBP_PC8 SBP_PC9 SBP_PC10' 
genoFile=`echo ~/UKBiobank/data/exome_files/project_VCF/072721_run/plink/ukb23156_c{1..22}.merged.filtered.bed`
bfile=~/UKBiobank/genotype_files_processed/010622_exp_white_finalqc/cache/*.bed
anno_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.anno_file
set_list=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.set_list_file
mask_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.mask_file
aaf_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.aff_file
build_mask=max
aaf_bins='0.005 0.01'
tpl_file=~/project/bioworkflows/admin/csg.yml
lmm_sos=~/project/bioworkflows/GWAS/LMM.ipynb
container_marp=~/containers/marp.sif
container_lmm=~/containers/lmm.sif 
lmm_job_size=1
ylim=0
k=10
reverse_log_p=True
numThreads=20
formatFile_regenie=~/project/UKBB_GWAS_dev/data/regenie_template.yml
bsize=1000
## Trait leave empty for qt traits
trait=
minMAC=1
snpannofile=~/UKBiobank/results/ukb23155_200Kexomes_annovar/2021_10_12_hg38_exome/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.gz

lmm_args="""regenie_burden
    --cwd $lmm_dir_regenie 
    --bfile $bfile 
    --genoFile $genoFile
    --phenoFile $phenoFile 
    --formatFile $formatFile_regenie 
    --phenoCol $phenoCol
    --covarCol $covarCol  
    --qCovarCol $qCovarCol
    --bsize $bsize
    --trait $trait
    --anno_file $anno_file
    --set_list $set_list
    --mask_file $mask_file
    --aaf_file $aaf_file
    --aaf_bins $aaf_bins
    --build_mask $build_mask
    --job_size $lmm_job_size
    --ylim $ylim
    --k $k
    --reverse_log_p $reverse_log_p
    --numThreads $numThreads
    --minMAC $minMAC
    --snpannofile $snpannofile
    --container_lmm $container_lmm
    --container_marp $container_marp
"""

sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \
    --template-file $tpl_file \
    --workflow-file $lmm_sos \
    --to-script $lmm_sbatch_regenie \
    --args "$lmm_args"

/home/dmc2245/miniconda3/lib/python3.8/site-packages/sos_notebook/kernel.py:1334: DeprecationWarning: Kernel._parent_header is deprecated in ipykernel 6. Use .get_parent()
  msg['msg_id'] = self._parent_header['header']['msg_id']


INFO: Running csg: Configuration for Columbia csg partition cluster
INFO: csg is completed.
INFO: csg output:   /home/dmc2245/pleiotropy/LMM/systolic_plus15inmedicated_200k_exomes-regenie-burden_2022-04-29.sbatch
INFO: Workflow csg (ID=w2300964bdfbac263) is executed successfully with 1 completed step.



# 04-29-22 Burden analysis with the raw diastolic_plus10inmedicated for extended white

In [3]:
lmm_dir_regenie=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/042922_diastolic_plus10inmedicated_extwhite
lmm_sbatch_regenie=~/pleiotropy/LMM/diastolic_plus10inmedicated_200k_exomes-regenie-burden_$(date +"%Y-%m-%d").sbatch
phenoFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_white_bloodpressure_invnorm_pcs_new
covarFile=~/UKBiobank/phenotype_files/pleiotropy/UKB_exome_white_bloodpressure_invnorm_pcs_new
phenoCol=diastolic_plus10inmedicated
covarCol=sex
qCovarCol='age Alcohol1 Alcohol2 Alcohol3 Alcohol4 smoking1 smoking2 DBP_PC1 DBP_PC2 DBP_PC3 DBP_PC4 DBP_PC5 DBP_PC6 DBP_PC7 DBP_PC8 DBP_PC9 DBP_PC10' 
genoFile=`echo ~/UKBiobank/data/exome_files/project_VCF/072721_run/plink/ukb23156_c{1..22}.merged.filtered.bed`
bfile=~/UKBiobank/genotype_files_processed/010622_exp_white_finalqc/cache/*.bed
anno_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.anno_file
set_list=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.set_list_file
mask_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.mask_file
aaf_file=~/UKBiobank/results_pleiotropy/REGENIE_results/results_burden_exome/102121_burden_files/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.aff_file
build_mask=max
aaf_bins='0.005 0.01'
tpl_file=~/project/bioworkflows/admin/csg.yml
lmm_sos=~/project/bioworkflows/GWAS/LMM.ipynb
container_marp=~/containers/marp.sif
container_lmm=~/containers/lmm.sif 
lmm_job_size=1
ylim=0
k=10
reverse_log_p=True
numThreads=20
formatFile_regenie=~/project/UKBB_GWAS_dev/data/regenie_template.yml
bsize=1000
trait=
minMAC=1
snpannofile=~/UKBiobank/results/ukb23155_200Kexomes_annovar/2021_10_12_hg38_exome/ukb23155_chr1_chr22_091321.hg38.hg38_multianno.renamedcols.csv.gz

lmm_args="""regenie_burden
    --cwd $lmm_dir_regenie 
    --bfile $bfile 
    --genoFile $genoFile
    --phenoFile $phenoFile 
    --formatFile $formatFile_regenie 
    --phenoCol $phenoCol
    --covarCol $covarCol  
    --qCovarCol $qCovarCol
    --bsize $bsize
    --trait $trait
    --anno_file $anno_file
    --set_list $set_list
    --mask_file $mask_file
    --aaf_file $aaf_file
    --aaf_bins $aaf_bins
    --build_mask $build_mask
    --job_size $lmm_job_size
    --ylim $ylim
    --k $k
    --reverse_log_p $reverse_log_p
    --numThreads $numThreads
    --minMAC $minMAC
    --snpannofile $snpannofile
    --container_lmm $container_lmm
    --container_marp $container_marp
"""

sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \
    --template-file $tpl_file \
    --workflow-file $lmm_sos \
    --to-script $lmm_sbatch_regenie \
    --args "$lmm_args"

INFO: Running csg: Configuration for Columbia csg partition cluster
INFO: csg is completed.
INFO: csg output:   /home/dmc2245/pleiotropy/LMM/diastolic_plus10inmedicated_200k_exomes-regenie-burden_2022-04-29.sbatch
INFO: Workflow csg (ID=w1c1654f1075d5a11) is executed successfully with 1 completed step.

